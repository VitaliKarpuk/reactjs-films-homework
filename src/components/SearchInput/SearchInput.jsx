import React, { useState } from "react";
import PropTypes from "prop-types";

const SearchInput = ({ onSearchFilms }) => {
  const [value, setValue] = useState("");
  const onChangeValue = e => {
    setValue(e.target.value);
  };

  const onHandleEnter = ({ key }) => {
    if (key === "Enter") {
      onSearchFilms(value);
    }
  };
  return (
    <div className="header__search">
      <input
        type="text"
        placeholder="the jungle book"
        value={value}
        onChange={onChangeValue}
        onKeyPress={onHandleEnter}
      />
      <span className="header__search-icon"></span>
    </div>
  );
};

SearchInput.propTypes = {
  onSearchFilms: PropTypes.func
};

SearchInput.defaultProps = {
  onSearchFilms: () => {}
};

export default SearchInput;
