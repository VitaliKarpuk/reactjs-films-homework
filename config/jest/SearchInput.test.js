/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import cofigureMockStore from 'redux-mock-store';
import SearchInput from '../../src/components/SearchInput/SearchInput';

const mockStore = cofigureMockStore();

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      films: [{ id: 1, genre_ids: [1] }],
      genre: [{ id: 1 }],
      dispatch: jest.fn()
    });


    component = renderer.create(
      <Provider store={store}>
        <SearchInput />
      </Provider>
    );
  });

  it("renders Search component", () => {
    expect(component.root.findByType('input').props.value).toEqual('')
    component.root.findByType('input').props.onChange({target: {value: 'test'}});
    expect(component.toJSON()).toMatchSnapshot();
    component.root.findByType('input').props.onKeyPress({key: "Enter"});
  });
})

// it("renders Search component", () => {
//   act(() => {
//     render(<SearchInput />, container);
//   });
//   const input = container.querySelector('input');
//   expect(input.value).toBe("");
//   ReactTestUtils.Simulate.change(input, { target: { value: 'test' } });
//   expect(input.value).toBe("test");
//   ReactTestUtils.Simulate.keyDown(input, { key: "E" });

// });


// it('Test event onKeyPress', () => {
//   let root = TestRenderer.create(<SearchInput />)
//   root.toJSON().children[0].props.onKeyPress({ key: "E" });
//   expect(root.toJSON()).toMatchSnapshot();
// })