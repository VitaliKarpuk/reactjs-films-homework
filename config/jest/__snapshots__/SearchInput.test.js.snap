// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders Search component 1`] = `
<div
  className="header__search"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="the jungle book"
    type="text"
    value="test"
  />
  <span
    className="header__search-icon"
  />
</div>
`;
